quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/legacy/infinispan/CacheManagerFactory.java

```diff
@@ -129,8 +140,46 @@ private void shutdownThreadPool() {
     private void configureTransportStack(ConfigurationBuilderHolder builder) {
         String transportStack = Configuration.getRawValue("kc.cache-stack");
 
+        var transportConfig = builder.getGlobalConfigurationBuilder().transport();
         if (transportStack != null && !transportStack.isBlank()) {
-            builder.getGlobalConfigurationBuilder().transport().defaultTransport().stack(transportStack);
+            transportConfig.defaultTransport().stack(transportStack);
         }
+
+        if (booleanProperty(JGROUPS_TLS_ENABLED_PROPERTY)) {
```

Comment:
Would it be correct to assume that TLS works only with TCP as a transport portocol? If yes, please add a check for that, and throw an exception if a mismatch is detected.

Reply:
Yes, this only works with TCP sockets. UDP sockets or Java NIO channels are not supported.

Reply:
I saw your latest changes for this. I suppose it works, still I wonder why this is doing a several negative checks (things that shouldn't be present) instead of a positive check that it is using TCP.